### PR TITLE
Added pop(), popitem(), __reversed__(); Documented unimplemented methods

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

### CODE REVIEW

**Comment by andy-maier** [comment]
Rebased to get coverage across all Python versions tested in the Travis run.

File: nocasedict/_nocasedict.py [hunk 1]
```diff
@@ -48,6 +48,10 @@
 
 __all__ = ['NocaseDict']
 
+# Used as default value for parameters to detect that it has not been
```
Comment (KSchopmeyer) [inline nocasedict/_nocasedict.py:51]: should be parameter (singular) or they (plural)

Comment (andy-maier) [reply nocasedict/_nocasedict.py]: DONE

**Comment by andy-maier** [comment]
I had to add `__reversed__()` to the _DictView class in order to support reversing the result of `keys()` etc.

**Pull request merged**